wily/commands/rank.py

```diff
@@ -0,0 +1,128 @@
+"""
+Rank command.
+
+The report command gives a table of files sorted according their ranking scheme
+of a specified metric.
+Will compare the values between files and return a sorted table.
+
+
+TODO: Refactor individual file work into a separate function
+TODO: Layer on Click invocation in operators section, __main__.py file
+TODO: Is the a better way to retrieve the revision number than via index?
+"""
+import tabulate
+import os
+import operator as op
+
+from wily import logger, format_date, format_revision, MAX_MESSAGE_WIDTH
+from wily.config import DEFAULT_GRID_STYLE
+from wily.operators import resolve_metric, MetricType
+from wily.state import State
+
+
+def rank(config, path, metric="maintainability.index", revision_number=0):
+    """
+
+    :param config: The configuration
+    :type config: :class:'wily.config.WilyConfig'
+
+    :param path: The path to the file
+    :type path ''str''
+
+    :param metric: Name of the metric to report on
+    :type metric: ''str''
+
+    :param revision_number: Version of git repository to revert to.
+    :type revision_number: ''int''
+
+    :return: Sorted table of all files in path, sorted in order of metric.
+    """
+    logger.debug("Running rank command")
+    logger.info(f"-----------Rank for {metric}------------")
+
+    data = []
+
+    operator, key = metric.split(".")
+    metric = resolve_metric(metric)
+    metric_meta = {
+        "key": key,
+        "operator": operator,
+        "title": metric.description,
+        "type": metric.type,
+        "wily_metric_type": metric.measure.name,  # AimHigh, AimLow, Informational
+        }
+
+    # vals = []
+    # Assumption is there is only one metric (e.g., therefore list of metrics commented out)
+    items = os.listdir(path)
```

Comment:
Use pathlib here instead of `os.listdir`. If you want a recursive search, use:

```python
pth = pathlib.Path(path)
items = pth.glob('**/*.py')
```

All the methods like `is_directory` exist, https://docs.python.org/3/library/pathlib.html

Reply:
I have the path recursive way of doing this in the latest commit

    pth = Path(path)
    items = pth.glob('**/*.py')